server/src/main/java/org/elasticsearch/cluster/routing/IndexRouting.java

```diff
@@ -220,10 +232,14 @@ public void collectSearchShards(String routing, IntConsumer consumer) {
         @Override
         public int indexShard(String id, @Nullable String routing, XContentType sourceType, BytesReference source) {
```

Comment:
Yeah. It's used in the other implementations of this interface. But there is a genuinely confusing thing here! We get an auto-generated ID even though we never use it. We don't use it for routing because we extract that from the source. And we don't use it for the _id itself because we extract that from the parsed dimensions. I mentioned that in a follow up but I think it's good to call out.

Reply:
sorry, I get the wrong line, the method that not use id is `private int readShard(String id, @Nullable String routing)`.
I just now comment in that line.

Reply:
Gotcha!